datahub-web-react/src/app/ingest/source/IngestedAssets.tsx

```diff
@@ -0,0 +1,174 @@
+import { Button, Typography } from 'antd';
+import React, { useState } from 'react';
+import { useGetSearchResultsForMultipleQuery } from '../../../graphql/search.generated';
+import { EntityType, FacetMetadata } from '../../../types.generated';
+import { EmbeddedListSearchModal } from '../../entity/shared/components/styled/search/EmbeddedListSearchModal';
+import { ANTD_GRAY } from '../../entity/shared/constants';
+import { formatNumber } from '../../shared/formatNumber';
+import { Message } from '../../shared/Message';
+import { capitalizeFirstLetterOnly, pluralize } from '../../shared/textUtil';
+
+const ENTITIES_WITH_SUBTYPES = new Set([
+    EntityType.Dataset.toLowerCase(),
+    EntityType.Container.toLowerCase(),
+    EntityType.Notebook.toLowerCase(),
+]);
+
+type Props = {
+    urn: string;
+    id: string;
+};
+
+type EntityTypeCount = {
+    count: number;
+    displayName: string;
+};
+
+/**
+ * Extract entity type counts to display in the ingestion summary.
+ *
+ * @param entityTypeFacets the filter facets for entity type.
+ * @param subTypeFacets the filter facets for sub types.
+ */
+const extractEntityTypeCountsFromFacets = (
```

Comment:
this nice pure function could be pretty easily testable! could do in a followup too

Reply:
agreed - i should probably also move this into some util file... will do!